utils/zstd/Makefile

```
@@ -66,6 +61,9 @@ define Package/zstd/description
 	This package provides the zstd binaries.
 endef
 
+TARGET_CFLAGS += -flto $(if $(CONFIG_ZSTD_OPTIMIZE_O3),$(filter-out -O%,$(TARGET_CFLAGS)) -O3,)
```

```suggestion
TARGET_CFLAGS := $(if $(CONFIG_ZSTD_OPTIMIZE_O3),$(filter-out -O%,$(TARGET_CFLAGS)) -O3,$(TARGET_CFLAGS)) -flto
```
I don't think you intend to **append** filtered `TARGET_CFLAGS` here, do you?  Keep in mind that using multiple `-O` options has a [defined behavior](https://gcc.gnu.org/onlinedocs/gcc/Optimize-Options.html), so you could just append `-O3`, if you don't think it _feels wrong_: :laughing: 
> If you use multiple -O options, with or without level numbers, the last such option is the one that is effective. 

```suggestion
TARGET_CFLAGS += -flto $(if $(CONFIG_ZSTD_OPTIMIZE_O3),-O3)
```

---

Simple enough

---

The reason why we filter out -O flags is to avoid confusion and potenital "breakage" (unintentional changes). Please don't remove this.

---

I certainly understand the "confusion" part--you must know the defined behavior, which may not be obvious--and that's why it was not my first choice, and we can leave it there.
However, I can't see how the omission of `filter-out` can lead to breakage or unintentional changes.

---

I added it back anyway.

---

@cotequeiroz 
If upstream devices to change on how CFLAGS are parsed?